### ISSUE TITLE
Petitioner/Practitioner: Review Petition before starting a case

### ISSUE THREAD

**Issue by mariahkannenberg** [flexion/ef-cms#1402]
As a Petitioner or Practitioner, in order to make sure that I've entered in accurate information as I'm filing my Petition, I need the ability to review all of my entries and documentation before starting a case.

## Pre-Conditions:
* Petitioner or Practitioner has entered all required information in the "Start a Case" form and is ready to submit it to the Court and start a new case

## Acceptance Criteria:
* Petitioner or Practitioner has the ability to review and change any information submitted on the "Start a Case" form
* Petitioner or Practitioner can view the actual PDF of each document to be transmitted to the Court as part of the review process (#1566 functionality)
* User should be able to perform this function on mobile

## Notes:
* Confirm design is done (posted below)

## Tasks
 - [x]  Create review view (Matt)
 - [x]  Routing (for back button use) (Matt)
 - [x]  Generate document previews (Matt)

## Definition of Done (Updated 8-15-19)

**Product Owner**
 - [x]  Acceptance criteria have been met

**UX**
 - [x] Usability has been validated
 - [x] Wiki has been updated (if applicable)

**Team**
 - [x] Test scenarios to meet all acceptance criteria have been written

**Engineering**
 - [x] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [x] New screens have been added to pa11y scripts
 - [x] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [x] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [x] Module dependencies are up-to-date and are at the latest resolvable version (npm update)
 - [x] Errors in Sonarcloud are fixed https://sonarcloud.io/organizations/flexion-github/projects
 - [x] Lambdas include CloudWatch logging of users, inputs and outputs
 - [x] Code refactored for clarity and to remove any known technical debt 
 - [x] Deployed to the dev environment
 - [x] Deployed to the stage environment

**Comment by klohman** [comment]
### Review Page : One Party 
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/5ac238ae-c51a-4c8b-8792-06e284958518)

* If ODS is not required, field is hidden 


### Review Page - Two Parties (Petitioner and Spouse/Deceased Spouse) 
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/33ac6a1f-74b2-407e-9348-5e1a0b331221)

**Comment by mariahkannenberg** [comment]
## Test Scenarios

**Petitioner Views Uploaded Document Before Submitting Case (File a Petition Form)** (out of scope - #1402)

Petitioner logs in
Petitioner selects File a Petition
Petitioner selects “Got it, Let’s Start My Case”
Petitioner uploads all documents 
Petitioner fills out all metadata associated with the case
Petitioner selects “Review and Sign” checkbox
Petitioner sees document name is a link
Petitioner selects document link and document opens in a new page
Petitioner selects “Submit to US Tax Court”
Petitioner logs out


**Practitioner Views Uploaded Document Before Submitting Case (File a Petition Form)** (out of scope - #1402)

Practitioner logs in
Practitioner selects File a Petition
Practitioner selects “Got it, Let’s Start My Case”
Practitioner uploads all documents 
Practitioner fills out all metadata associated with the case
Practitioner selects “Review and Sign” checkbox
Practitioner selects “Submit to US Tax Court”
Practitioner logs out
IRS Attorney Uploads Document To Existing Case
IRS Attorney logs in
Practitioner selects File a Petition
Practitioner selects “Got it, Let’s Start My Case”
Practitioner uploads all documents 
Practitioner fills out all metadata associated with the case
Practitioner selects “Review and Sign” checkbox
Practitioner sees document name is a link
Practitioner selects document link and document opens in a new page
Practitioner selects “Submit to US Tax Court”
Practitioner logs out

**Comment by mariahkannenberg** [comment]
@klohman Add mobile design.

**Comment by matthopson** [comment]
@sarahridgway Question about the petitioner information labels. I'm seeing:

`Contact Information` for "Partnership",
`Your Contact Information` and `Spouse’s Contact Information` for "Myself and my spouse"

Do we have these label rules documented somewhere?

**Comment by JessicaMarine** [comment]
PO Review 8/23/19:
* When I try to preview any of the document(s) from Step 5 of 5 on a mobile device, I get a screen that says "There was an error loading the document preview. Please try again."  I tried with several different documents, all with the same result.
